src/components/experiments/single-view/results/AggregateRecommendationDisplay.tsx

```
@@ -0,0 +1,40 @@
+import React from 'react'
+
+import { AggregateRecommendation, AggregateRecommendationType } from 'src/lib/analyses'
+import { ExperimentFull } from 'src/lib/schemas'
+
+/**
+ * Displays an AggregateRecommendation.
+ */
+export default function AggregateRecommendationDisplay({
+  aggregateRecommendation,
+  experiment,
+}: {
+  aggregateRecommendation: AggregateRecommendation
+  experiment: ExperimentFull
+}): JSX.Element {
+  switch (aggregateRecommendation.type) {
+    case AggregateRecommendationType.ManualAnalysisRequired:
+      return <>Manual Analysis Required</>
+    case AggregateRecommendationType.NotAnalyzedYet:
+      return <>Not Analyzed Yet</>
+    case AggregateRecommendationType.Inconclusive:
+      return <>Inconclusive</>
+    case AggregateRecommendationType.DeployEither:
+      return <>Deploy Either Variation</>
+    case AggregateRecommendationType.Deploy: {
+      const chosenVariation = experiment.variations.find(
+        (variation) => variation.variationId === aggregateRecommendation.variationId,
+      )
+      // istanbul ignore next; Typeguard
+      if (!chosenVariation) {
+        throw new Error('No match for chosenVariationId among variations in experiment.')
+      }
+
+      return <>Deploy {chosenVariation.name}</>
+    }
+    // istanbul ignore next; Shouldn't occur
+    default:
+      throw new Error('Missing AggregateRecommendationType.')
```

Is this needed at all? Doesn't TypeScript verify that the `case`s cover all options?

---

Typescript will not ensure that the `case`s cover all options. 
This would fail silently if `AggregateRecommendationType` was extended and there was no default case.

---

Why not test it then? :slightly_smiling_face: